app/vmctl/vm_native.go

```diff
@@ -54,13 +59,75 @@ const (
 	nativeExportAddr  = "api/v1/export/native"
 	nativeImportAddr  = "api/v1/import/native"
 	nativeTenantsAddr = "admin/tenants"
+	nativeSeriesAddr  = "api/v1/series"
 
 	nativeBarTpl = `Total: {{counters . }} {{ cycle . "↖" "↗" "↘" "↙" }} Speed: {{speed . }} {{string . "suffix"}}`
 )
 
 func (p *vmNativeProcessor) run(ctx context.Context) error {
+	if p.cc == 0 {
+		p.cc = 1
+	}
+
+	fmt.Printf("Init series discovery process on time range %s - %s \n", p.filter.timeStart, p.filter.timeEnd)
+	series, err := p.explore(ctx, p.filter)
+	if err != nil {
+		return fmt.Errorf("cannot get series from source %s database: %s", p.src.addr, err)
+	}
+	fmt.Printf("Discovered %d series \n", len(series))
+
+	pool := pb.NewPool()
+	if err := pool.Start(); err != nil {
+		log.Printf("error start process bars pool: %s", err)
+		return err
+	}
+	bar := pb.ProgressBarTemplate(nativeBarTpl).New(0)
+	pool.Add(bar)
+	defer func() {
+		bar.Finish()
+		if err := pool.Stop(); err != nil {
+			fmt.Printf("failed to stop barpool: %+v\n", err)
+		}
+	}()
+
+	requestLimitC := make(chan struct{}, p.requestsLimit)
+	filterCh := make(chan filter)
+	var wg sync.WaitGroup
+
+	defer func() {
+		close(filterCh)
+		wg.Wait()
+		close(requestLimitC)
+	}()
+
```

Comment:
These changes seems irrelevant to the PR purpose

Reply:
will add this changes in the next improvement